x-pack/plugins/security_solution/public/notes/components/add_note.tsx

```diff
@@ -88,11 +94,22 @@ export const AddNote = memo(
           },
         })
       );
+
+      // Automatically pin an associated event if it's attached to a timeline and it's not pinned yet
```

Comment:
would you be ok with passing a callback as props instead than having the code here? Something like `onNoteAdded`. The reason is when I created those components, I made sure (as much as possible) that they would not be anything timeline-specific code (as these are notes and shouldn't  really know about timeline, alerts...)

I understand that there are a couple of places in this file called `timelineId` but that is just because currently the api is using that word. Ideally we would want to rename all those variables in the frontend by `savedObjectId` as a note can be associated with pretty much any `savedObjectId`, not just timeline ones...

Does that makes sense?

Reply:
I think it is a great feedback but i have a question on a separate scenario.

Consider below scenario: 
1. If I add a note on an alert in AlertTable.
2. Then I add that event to a timeline. 

Now since that event is attached to the timeline, should it already be pinned to that timeline?

Reply:
nope because in your scenario, you created a note attached to a document only. Having an event in a timeline doesn't mean that all its note will be associated with that timeline. You can still create note NOT associated with the timeline by unchecking the `Attach to timeline` checkbox.
The pin icon only reflects note that have been associated with a timelineId.

Does that make sense?

Reply:
It does make sense. In that case, do you think we should be more clear with the modal message which says `Event cannot be unpinned because there are notes attached`. It should rather say something along the lines of `Event cannot be unpinned because this event and timeline have notes attached`.

Not a big issue but something to keep a note of.

Reply:
yes that makes sense, we can get with @nastasha-solomon for the proper wording. Something like `This event cannot be unpinned because it has notes associated with this current timeline`.

Reply:
Slightly modified the tooltip to `This event cannot be unpinned because it has notes in timeline`

Reply:
I think explaining why an event or alert can't be unpinned from a Timeline is useful, but I'm also wondering if it'd be _more_ helpful for users to know what to do if they truly want to unpin the event or alert from Timeline. What do you think @christineweng?

Reply:
@nastasha-solomon in this case, if users truly want to unpin the event, they will have to delete the note

Reply:
What do you think about this: `Delete this event's notes to unpin it from Timeline.`

If you need copy for pinned alerts, you can exchange "event's" with "alert's".

Reply:
a note can be tied to a document, or timeline, or both.. `Delete this event's notes related to this timeline to unpin it from this Timeline.` is more accurate

also, it is possible to have notes associated to different timelines... users will need to know which one to delete. to me the additional message is encouraging users to delete notes just to unpin... not sure we want that though. wdyt? @PhilippeOberti @paulewing 

<img width="1052" alt="image" src="https://github.com/user-attachments/assets/ed0f1954-3967-4136-92a8-47fad430de68">

Reply:
Yeah to let the user know exactly what to do, the message would have to be long and not necessary clear...
In an ideal world we could have a tooltip that explains what to do with a link to the timeline notes tab filtered specifically to display only the notes in question. This could be feasible with some frontend filtering option, but
- we don't really have the time to do that
- It might not be worth doing if timeline is going away shortly 
It would be a need enhancement of the notes tab though!

Reply:
I don't really have a good answer at this time. I'll think about it some more!

Reply:
Ah, ok I gotcha now. From the flyout (at least from now), there's no easy way to tell when a note is associated with a specific Timeline. Adding more text into the tooltip wouldn't necessarily help with that. Users would still need to go to the Notes management page to determine which notes are associated with certain Timelines. 

One possibility is to move forward with the original language (`This event cannot be unpinned because it has notes associated with a Timeline`) and link to the docs. Example:

`This event cannot be unpinned because it has notes associated with a Timeline. Learn more`

It's not the best UX, but it might be a better experience than forcing users to read a multi-sentence tooltip message.

Reply:
I think this is the best we can do with the time we have right now yes. Thank you @nastasha-solomon

Reply:
Connected with @nastasha-solomon offline, we are going with `This event cannot be unpinned because it has notes in Timeline.` at this time.

When the docs issue is available, consider adding a link at the end: `This event cannot be unpinned because it has notes in Timeline. Learn more` ([ticket here](https://github.com/elastic/kibana/issues/193495))